src/plots/plots.js

```diff
@@ -831,8 +776,60 @@ plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLa
 
         ax._anchorAxis = ax.anchor === 'free' ?
             null :
-            Plotly.Axes.getFromId(mockGd, ax.anchor);
+            axisIDs.getFromId(mockGd, ax.anchor);
+    }
+
+    for(i = 0; i < axList.length; i++) {
```

Comment:
> Any reason for putting this here instead of e.g. `Cartesian.plot`?
>
> I'm thinking this has to do with being compatible with `editType: 'ticks'`, but I want to confirm.

Actually I'm not entirely sure. `_mainSubplot` felt structural to me, in a way that I wanted it available sooner rather than later, but it's possible it could be deferred to `Cartesian.plot`.